allergies/allergies.js

```diff
@@ -0,0 +1,66 @@
+//empty arrays for the user inputs to go into
+var healthRestrictions = [];
+var dietRestrictions = [];
+
+/*click event for all elements with the "healthIcon" class, if clicked it updates the class to 
+ reflect that, changes the image to a purple version as needed, and updates the healthRestrictions arrays*/ 
+$('.healthIcon').on('click', function() {
+    if ($(this).hasClass('notClicked')) {
+
+        healthRestrictions.push($(this).attr('value'));
+
+        $('#' + $(this).attr('value') + 'Icon').attr('src', '../assets/purp icons/' + $(this).attr('value') + '.png');
```

Comment:
this is mentally a little difficult to parse, you may want to consider using additional variables like

```
var iconId = $(this).attr('value') + 'Icon';
$('#' + iconId)
```